Source/WebCore/platform/graphics/skia/ComplexTextControllerSkia.cpp

```diff
@@ -151,28 +151,41 @@ static std::optional<HBRun> findNextRun(std::span<const UChar> characters, unsig
     return std::optional<HBRun>({ startIndex, textIterator.currentIndex(), currentScript.value() });
 }
 
+static inline void forEachHBRun(const std::span<const UChar>& characters, bool ltr, std::function<void(const HBRun&)> callback)
+{
+    size_t offset = 0;
+    if (ltr) {
+        while (offset < characters.size()) {
+            auto run = findNextRun(characters, offset);
+            if (!run)
+                break;
+            offset = run->endIndex;
+            callback(*run);
+        }
+    } else {
+        Vector<HBRun> runList;
```

Comment:
We could save heap allocations here by using an inline size for the Vector, not sure which size, though.